src/app/services/EloCalculationService.ts

```diff
@@ -0,0 +1,106 @@
+                                                                        normalEloDifferences.pointChangeForOne);
+        return {
+            playerZero: {
+                currentElo: zeroNewElo,
+                numberOfGamePlayed: eloEntry.eloInfoPair.playerZero.numberOfGamePlayed + 1,
+            },
+            playerOne: {
+                currentElo: oneNewElo,
+                numberOfGamePlayed: eloEntry.eloInfoPair.playerOne.numberOfGamePlayed + 1,
+            },
+        };
+    }
+    public static getNormalEloDifferences(eloEntry: EloEntry): EloDifferences {
+        const wZero: number = EloCalculationService.getWFrom(eloEntry.winner, Player.ZERO);
+        const wOne: number = EloCalculationService.getWFrom(eloEntry.winner, Player.ONE);
+        const playerZeroInfo: EloInfo = eloEntry.eloInfoPair.playerZero;
+        const playerOneInfo: EloInfo = eloEntry.eloInfoPair.playerOne;
+        const eloZero: number = playerZeroInfo.currentElo;
+        const eloOne: number = eloEntry.eloInfoPair.playerOne.currentElo;
+        const pZero: number = this.getWinningProbability(eloZero, eloOne);
+        const pOne: number = this.getWinningProbability(eloOne, eloZero);
+        const playerZeroK: number = EloCalculationService.getKFrom(playerZeroInfo.numberOfGamePlayed);
+        const playerOneK: number = EloCalculationService.getKFrom(playerOneInfo.numberOfGamePlayed);
+        return {
+            pointChangeForZero: EloCalculationService.getNormalEloDifference(playerZeroK, wZero, pZero),
+            pointChangeForOne: EloCalculationService.getNormalEloDifference(playerOneK, wOne, pOne),
+        };
+    }
+    public static getNormalEloDifference(K: number, W: number, P: number): number {
+        return K * (W - P);
+    }
+    public static getActualNewElo(oldElo: number, eloDifference: number): number {
+        let minimumFinalElo: number = 1;
+        if (eloDifference < 0) {
+            if (oldElo < 100) {
+                eloDifference = 0; // Not removing elo to player bellow 100
+            } else {
+                minimumFinalElo = 100; // Not making a player fall back bellow 100
+            }
+        }
+        const newElo: number = oldElo + eloDifference;
+        return Math.max(minimumFinalElo, newElo);
+    }
+    public static getWFrom(winner: 'ZERO' | 'ONE' | 'DRAW', player: Player): number {
+        if (winner === 'DRAW') {
+            return 0.5;
+        } else if (winner === 'ZERO') {
+            if (player === Player.ZERO) {
+                return 1;
+            } else {
+                return 0;
+            }
+        } else {
+            Utils.expectToBe(winner, 'ONE');
+            if (player === Player.ONE) {
+                return 1;
+            } else {
+                return 0;
+            }
+        }
+    }
+    public static getKFrom(numberOfGamePlayed: number): number {
```

Comment:
what is K ? (celui-là j'ai compris à partir de tes explications, mais faut pas croire qu'un lecteur de ce fichier connais K !)

Reply:
expliqué dans getNormalEloDifference  ou les trois sont des paramètres, tu veux ça partouuut partouuut ?

Reply:
là où c'est relevant, une petite doc pour chaque méthode c'est sympa; une petite doc pour chaque méthode publique ça devrait être requis !